staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -219,6 +221,24 @@ func (a *AdmissionOptions) Validate() []error {
 			registeredPlugins.Difference(intersections).List()))
 	}
 
+	// TODO: is there a better location? we need to ensure that the
+	// admission plugin is not disabled by the admin when the feature
+	// gate is enabled, otherwise we skip the check whether the user
+	// has permission to do unsafe deletion of corrupt object
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AllowUnsafeMalformedObjectDeletion) {
+		// the plugin and the feature have the same name
+		// the plugin is enabled by default, and is active only
+		// when the feature is enabled
+		name := string(genericfeatures.AllowUnsafeMalformedObjectDeletion)
+		// developer error, the plugin should be registered
+		if !registeredPlugins.Has(name) {
+			errs = append(errs, fmt.Errorf("the plugin %q must be registered, and enabled by default", name))
+		}
+		if disablePlugins.Has(name) {
+			errs = append(errs, fmt.Errorf("the plugin %q must not be disabled when feature %q is enabled", name, name))
+		}
```

Comment:
do we have similar checks for other feature gates? I thought that configuring admission is highly discouraged, but technically fine to do.

Reply:
we are moving admission to the REST layer